courses/advanced-ada/chapters/generics.rst

```diff
@@ -1084,19 +1086,335 @@ In this example, we declare the array :ada:`A` with elements of both
 we call the :ada:`Average` function from :ada:`Ops`, an instance of the
 generic package :ada:`Float_Interface_Pkg.Ops`.
 
-Discussion: Generic interfaces vs. other approaches
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Discussion: formal interfaces vs. other approaches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:code-config:`reset_accumulator=True`
+
+In Ada, we basically have three approaches to describe interfaces. In
+addition to the approach using formal interfaces that we've just seen
+above, we also have these approaches:
+
+- Formal subprograms, which we've presented in the introductory course
+  (:doc:`../../intro-to-ada/chapters/generics`).
+
+- Formal packages, which we've discussed in the section on
+  :ref:`formal packages <FormalPackages>`).
```

Comment:
I went to review the section on Formal packages a bit more, and there is some incorrect information here: A signature package is a package with a package without a specification. So the package that you use as an example there is not really a signature package.

The idea of a signature package is that of an interface really. Even though we could expand the terminology, and putting some helpers in the specs is very useful in the real world (akin to default implementations in interfaces), the basic concept should be clarified IMHO.

Reply:
Your comment about signature packages was actually referring to content that was not in this branch. So I rebased this branch, created a separate section to elaborate on signature packages and adapted other parts of the chapter accordingly.